pkg/broadcaster/broadcaster.go

```diff
@@ -0,0 +1,92 @@
+package broadcaster
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+// Broadcaster is a simplified Broker allowing clients to subscribe while
+// itself behaving as a io.Writer.
+// The Broadcaster is itself a io.Writer, any write to it will be sent to
+// all the client that called .Subscribe() (or .UnsafeSubscribe()).
+//
+// Caveats:
+//   - The order of writes is not guaranteed.
+//   - Each write sequencially calls .Write() on the clients, if any client blocks
+//     or slows down, it impacts every client as well as the initial caller using Write()
+//     on the broadcaster.
+type Broadcaster struct {
+	mu sync.Mutex
+	// NOTE: As we use a map, the broadcast order is randomized.
+	// Consider using a slice instead to be deterministic.
+	clients map[io.WriteCloser]struct{}
```

Comment:
Document that clients==nil means that the Broadcaster is closed?

Reply:
Done.